src/qibolab/instruments/rfsoc/driver.py

```diff
@@ -200,8 +201,15 @@ def play(
                         i_pulse, q_pulse, qubits[ro_pulse.qubit]
                     )
                     if execution_parameters.averaging_mode is AveragingMode.CYCLIC:
-                        discriminated_shots = np.mean(discriminated_shots, axis=0)
-                    result = execution_parameters.results_type(discriminated_shots)
+                        _, counts = np.unique(
+                            discriminated_shots, return_counts=True, axis=0
+                        )
+                        freqs = counts / np.shape(discriminated_shots)[0]
```

Comment:
Why not?
```suggestion
                        freqs = counts / discriminated_shots.shape[0]
```

Isn't it always a `np.ndarray` object?